### PR TITLE
Add integration test for --fail-fast

### PR DESCRIPTION
**Pull request by oehme**

### CODE REVIEW

**Comment by michael-o** [comment]
Will add and run it!

**Comment by michael-o** [comment]
I do not understand the test.:

* You are not even passing `--fail-fast`. Did you forget that?
* You make sure that `Module3` will never run because `Module1` will fail, `Module2` needs to be built first because `Module3` relies on it?

File: core-it-suite/src/test/java/org/apache/maven/it/MavenITmng6720FailFastTest.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        }
+        verifier.resetStreams();
+
+        List<String> module1Lines = verifier.loadFile(new File(testDir, "module-1/target/surefire-reports/Module1Test-output.txt"), false);
+        assertTrue(module1Lines.contains("Module1"));
+        List<String> module2Lines = verifier.loadFile(new File(testDir, "module-2/target/surefire-reports/Module2Test-output.txt"), false);
+        assertTrue(module2Lines.contains("Module2"));
+        List<String> module3Lines = verifier.loadFile(new File(testDir, "module-3/target/surefire-reports/Module3Test-output.txt"), false);
+        assertTrue(module3Lines.isEmpty());
+
```
Comment (michael-o) [inline core-it-suite/src/test/java/org/apache/maven/it/MavenITmng6720FailFastTest.java]: Please check the formatting of these asserts...

File: core-it-suite/src/test/resources/mng-6720-fail-fast/module-1/src/test/java/Module1Test.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import org.junit.Test;
+
+public class Module1Test {
+    @Test
+    public void test() throws Exception {
+        Thread.sleep(1000);
+        System.out.println("Module1");
+        throw new RuntimeException();
+    }
```
Comment (michael-o) [inline core-it-suite/src/test/resources/mng-6720-fail-fast/module-1/src/test/java/Module1Test.java]: Formatting

File: core-it-suite/src/test/resources/mng-6720-fail-fast/module-2/src/test/java/Module2Test.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import org.junit.Test;
+
+public class Module2Test {
+    @Test
+    public void test() throws Exception {
+        Thread.sleep(2000);
+        System.out.println("Module2");
+    }
+}
```
Comment (michael-o) [inline core-it-suite/src/test/resources/mng-6720-fail-fast/module-2/src/test/java/Module2Test.java]: Formatting

File: core-it-suite/src/test/resources/mng-6720-fail-fast/module-3/src/test/java/Module3Test.java [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import org.junit.Test;
+
+public class Module3Test {
+    @Test
+    public void test() {
+        System.out.println("Module3");
+    }
+}
```
Comment (michael-o) [inline core-it-suite/src/test/resources/mng-6720-fail-fast/module-3/src/test/java/Module3Test.java]: Formatting

**Comment by oehme** [comment]
> You are not even passing --fail-fast. Did you forget that?

--fail-fast is the default

> You make sure that Module3 will never run because Module1 will fail, Module2 needs to be built first because Module3 relies on it?

Yep

**Comment by michael-o** [comment]
@rfscholte . This looks good to me. It properly passes on my Windows 10 desktop, an old, low-perf 4-thread Atom CPU with FreeBSD 11.2-RELEASE and a 20-thread Xeon server with FreeBSD 12-STABLE. I'd ready to merge this into master along with the Maven change.

**Comment by michael-o** [comment]
I will now also try the negative case with Maven from master.

**Comment by michael-o** [comment]
Very good, with Maven from master:

    mng6720FailFast(ItShouldWaitForConcurrentModulesToFinish)...FAILURE (3.1 s)

```
[INFO] Skipping parent
[INFO] This project has been banned from the build due to previous failures.
[INFO] module-1 ........................................... FAILURE [  2.905 s]
[INFO] ------------------------------------------------------------------------
[INFO] module-2 ........................................... SKIPPED
[INFO] module-3 ........................................... SKIPPED
[INFO] parent ............................................. SKIPPED
[INFO] ------------------------------------------------------------------------
```

**Comment by olamy** [comment]
IT test for https://github.com/apache/maven/pull/272

File: core-it-suite/src/test/java/org/apache/maven/it/MavenITmng6720FailFastTest.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+ * <a href="https://issues.apache.org/jira/browse/MNG-6720">MNG-6720</a>.
+ *
+ */
+public class MavenITmng6720FailFastTest
+    extends AbstractMavenIntegrationTestCase
+{
+
+    public MavenITmng6720FailFastTest()
+    {
+        super( "[3.6.0-SNAPSHOT,)" );
```
Comment (rfscholte) [inline core-it-suite/src/test/java/org/apache/maven/it/MavenITmng6720FailFastTest.java]: This should be [3.6.2,)

Comment (michael-o) [reply core-it-suite/src/test/java/org/apache/maven/it/MavenITmng6720FailFastTest.java]: Fixed, but what about these:

```
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng3743ForkWithPluginManagementTest.java:        super( "(2.0.8,3.1-SNAPSHOT)" ); // only test in 2.0.9+
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng5771CoreExtensionsTest.java:        requiresMavenVersion( "[3.3.2-SNAPSHOT,)" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng5965ParallelBuildMultipliesWorkTest.java:        super( "[3.6.1-SNAPSHOT,)" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng6352PrintVersionTest.java:        super( "[3.5.3-SNAPSHOT,3.5.4-SNAPSHOT]" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng6506PackageAnnotationTest.java:        super( "[3.6.1-SNAPSHOT,)" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng6558ToolchainsBuildingEventTest.java:        super( "[3.6.1-SNAPSHOT,)" );
```

**Comment by oehme** [comment]
Am I correct that this has been merged, so the PR can be closed now?

**Comment by olamy** [comment]
@oehme some changes have been requested :)

**Comment by oehme** [comment]
Yeah, but @michael-o already did those changes and committed here: https://github.com/apache/maven-integration-testing/commit/3ccccc4cc3e4c7276496118e775dc68e631a3081

**Comment by olamy** [comment]
merged via https://github.com/apache/maven-integration-testing/commit/3ccccc4cc3e4c7276496118e775dc68e631a3081

**Comment by michael-o** [comment]
I have now merge into master. @olamy the commit existed on a branch only. No issue. All is fine now.

**Pull request closed without merging**